lib/monaco-adapter.js

```diff
@@ -23,10 +23,10 @@ firepad.MonacoAdapter = (function() {
   };
 
   MonacoAdapter.prototype.detach = function() {
-    this.monaco.onDidChangeModelContent(this.onChange).dispose();
-    this.monaco.onDidBlurEditor(this.onBlur).dispose();
-    this.monaco.onDidFocusEditor(this.onFocus).dispose();
-    this.monaco.onDidChangeCursorPosition(this.onCursorActivity).dispose();
+    this.onChange.dispose();
+    this.onBlur.dispose();
+    this.onFocus.dispose();
+    this.onCursorActivity.dispose();
```

Comment:
Hrm...  Sorry to keep bugging you about this code, but have you tried calling firepad.dispose() to verify this actually runs okay?  Unless calling `this.monaco.onDidChangeModelContent(this.onChange)` actually mutates your onChange function and adds a dispose property, I'd guess that this will not work correctly.

I would guess that what you need to do is something like:

```
var changeHandler = this.monaco.onDidChangeModelContent(this.onChange);
var didBlurHandler = this.monaco.onDidBlurEditor(this.onBlur);
...

this.detach = function() {
  changeHandler.dispose();
  blurHandler.dispose();
  ...
}
```

Reply:
No worries, Thank you for helping me out with this. I'm clueless about the detach function.